Boom boom

Beaverperson opened the pull request.

Beaverperson committed:
Here is my first commit name

ivannizh commented on main.go:
```diff
@@ -97,9 +97,12 @@ func (s ParcelService) Delete(number int) error {
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
```
После проверки ошибок БД нужно закрыть через дефер

ivannizh commented on parcel.go:
```diff
@@ -13,48 +14,108 @@ func NewParcelStore(db *sql.DB) ParcelStore {
+			return nil, err
+		}
+		res = append(res, bufferPars)
+	}
```
После цикла нужно проверить курсор на наличие ошибок

http://go-database-sql.org/retrieving.html

ivannizh commented on parcel.go:
```diff
@@ -13,48 +14,108 @@ func NewParcelStore(db *sql.DB) ParcelStore {
-	// реализуйте обновление адреса в таблице parcel
-	// менять адрес можно только если значение статуса registered
-
+	targetParcel, err := s.Get(number)
```
В этой и следующей функции нужно упростить логику и делать все в одном запросе.

Нужно убрать первый запрос и добавить в `UPDATE`(`DELETE`) еще одно условие в секцию `WHERE`

ivannizh commented on parcel_test.go:
```diff
@@ -30,59 +31,82 @@ func getTestParcel() Parcel {
-	db, err := // настройте подключение к БД
+
+	db, err := sql.Open("sqlite", "tracker.db")
+	require.NoError(t, err)
```
Так же после проверки ошибок нужно закрыть БД

ivannizh replied:
Нужно исправить во всех тестах

ivannizh commented on parcel_test.go:
```diff
@@ -98,24 +122,24 @@ func TestGetByClient(t *testing.T) {
+		targetParcel, inMap := parcelMap[parcel.Number]
+		require.True(t, inMap)
+		targetParcel.Number = parcel.Number
+		require.Equal(t, parcel, targetParcel)
```
На втором месте нужно ставить что хотим увидеть, а на третьем, что проверяем.
На самом тест это не влияет, но когда он упадет, то вывод будет неправильный

ivannizh reviewed (commented):
Привет! Хорошая работа! Осталось исправить некоторые моменты и работа будет принята.

Beaverperson committed:
Add DB connection closure after use
Rows iteration error handling
Correct parcel status check for update now made via SQL check

ivannizh commented on main.go:
```diff
@@ -140,36 +141,36 @@ func main() {
 	if err != nil {
 		fmt.Println(err)
-		return
+		//return
```
А зачем это было сделано? Нужно все вернуть на место, если ошибка, то на выход

ivannizh reviewed (commented):
Привет!

Все ок, кроме комментирования `return`, нужно его вернуть.

Beaverperson committed:
enable return in main.go

ivannizh commented on parcel_test.go:
```diff
@@ -75,7 +75,7 @@ func TestSetAddress(t *testing.T) {
 	targetParcel, err := store.Get(number)
 	require.NoError(t, err)
-	require.Equal(t, newAddress, targetParcel.Address)
+	require.Equal(t, targetParcel.Address, newAddress)
```
Было как раз правильно везде

ivannizh reviewed (commented):
Не увидел исправления `return` во всем `main`

Beaverperson committed:
why without hat

Beaverperson committed:
fix

ivannizh reviewed (approved):
Отлично, принято!
